src/Traversals/dfs.jl

```diff
@@ -2,52 +2,67 @@ import Base.showerror
 struct CycleError <: Exception end
 Base.showerror(io::IO, e::CycleError) = print(io, "Cycles are not allowed in this function.")
 
-struct DFS <: TraversalAlgorithm end
+struct DepthFirst <: TraversalAlgorithm
+    neighborfn::Function
+end
+
+DepthFirst(;neighborfn=outneighbors) = DepthFirst(neighborfn)
```

Comment:
the change above will make the keyword constructor type-unstable I think, because there is no dispatch on kwargs

Reply:
I think that's ok. Type instability in a constructor will not be a huge performance hit, will it?

Reply:
yes it's really fine, even more as there is another constructor available